service/controller/resource/tlscleanup/resource.go

```diff
@@ -0,0 +1,57 @@
+// tlscleanup provides a resource that cleans up unnecessary TLS certificates
+// from the shard namespace. These were created by earlier versions but are now
+// unused so we ensure to remove them to not leave them dangling.
+//
+// TODO: this resource can be removed after a couple of releases
```

Comment:
I would change this to
```suggestion
// TODO: this resource can be removed in next release.
```